vortex-array/src/canonical.rs

```diff
@@ -242,29 +242,37 @@ fn varbin_to_arrow(varbin_array: VarBinArray) -> ArrayRef {
     // Switch on Arrow DType.
     match varbin_array.dtype() {
         DType::Binary(_) => match offsets.ptype() {
-            PType::I32 => Arc::new(BinaryArray::new(
-                as_offset_buffer::<i32>(offsets),
-                data.into(),
-                nulls,
-            )),
-            PType::I64 => Arc::new(LargeBinaryArray::new(
-                as_offset_buffer::<i64>(offsets),
-                data.into(),
-                nulls,
-            )),
+            PType::I32 => Arc::new(unsafe {
```

Comment:
I'm not sure how much this actually saves us, once I get to ~parity with arrow numbers I'll turn the validations back on and see what difference it makes